pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -463,6 +463,7 @@ func (configgen *ConfigGeneratorImpl) buildGatewayHTTPRouteConfig(node *model.Pr
 			for _, hostname := range intersectingHosts {
 				if svc, exists := nameToServiceMap[hostname]; exists {
 					gatewayService = svc
+					break
```

Comment:
maybe i missed some ctx, wondering why donot you get gatewayService in the below loop L477.

So you can get one gatewayService per intersectingHosts

Reply:
yes, That can be done